### PR TITLE
Simplify `standaloneServer` API (v4)

### LINKED ISSUE #6483: Reduce `standaloneServer` steps (v4)
While pairing, @glasser and I noticed we can simplify the `standaloneServer` API a bit.

Instead of having the user call and `await` `listen`, we can do that within the `standaloneServer` function instead by accepting `listen` args and returning the result of calling `listen`.

so...

```ts
const { url } = await standaloneServer(apolloServer, listenOpts, serverOpts);
```

...or something to that effect.

(rather than)
```ts
const expressServer = standaloneServer(apolloServer, serverOpts);
const { url } = await expressServer.listen(listenOpts);
```

cc @rkoron007

### PR DESCRIPTION
**Pull request by trevor-scheer**
Fixes #6483

### CODE REVIEW

File: packages/server/src/standalone/index.ts [hunk 1]
```diff
@@ -14,20 +14,25 @@ interface HTTPServerOptions<TContext extends BaseContext> {
...
+  listenOptions: ListenOptions = { port: 4000 },
+): Promise<{ url: string }> {
   const context = options?.context ?? (async () => ({} as TContext));
-  return new ApolloServerStandalone<TContext>(server, { context });
+  return await new ApolloServerStandalone<TContext>(server, { context }).listen(
+    listenOptions,
+  );
 }
 
 class ApolloServerStandalone<TContext extends BaseContext> {
```
Comment (glasser) [inline packages/server/src/standalone/index.ts:38]: Do we still need the class at all?

File: packages/server/src/standalone/index.ts [hunk 2]
```diff
@@ -14,20 +14,25 @@ interface HTTPServerOptions<TContext extends BaseContext> {
   context?: ContextFunction<[ExpressContext], TContext>;
 }
```
Comment (glasser) [inline packages/server/src/standalone/index.ts:15]: Maybe ListenOptions should be a field on this rather than a third argument?

**Review by glasser — CHANGES_REQUESTED** [review]

File: packages/server/src/__tests__/standalone/standaloneSpecific.test.ts [hunk 3]
```diff
@@ -8,7 +8,8 @@ describe('Typings: TContext inference', () => {
     });
 
     // HTTPApolloServer<BaseContext>
-    standaloneServer(server);
+    await standaloneServer(server, {}, { port: 0 });
```
Comment (glasser) [inline packages/server/src/__tests__/standalone/standaloneSpecific.test.ts:11]: the idea being

`await standaloneServer(server, {listen: {port: 0}})`

File: packages/server/src/standalone/index.ts (hunk 1 above)
Comment (trevor-scheer) [reply packages/server/src/standalone/index.ts]: Nope, good call!

**Review by glasser — APPROVED** [review]

**Pull request merged**